com.unity.netcode.gameobjects/Runtime/Core/SnapshotSystem.cs

```diff
@@ -137,25 +152,44 @@ public int AddEntry(in VariableKey k)
             return pos;
         }
 
-        internal void AddSpawn(SnapshotSpawnCommand command)
+        internal List<ulong> GetClientList()
         {
-            if (NumSpawns < k_MaxSpawns)
+            List<ulong> clientList;
+            clientList = new List<ulong>();
+
+            if (!NetworkManager.IsServer)
             {
-                command.TargetClientIds = new List<ulong>();
-                if (!m_NetworkManager.IsServer)
-                {
-                    command.TargetClientIds.Add(m_NetworkManager.ServerClientId);
-                }
-                else
+                clientList.Add(NetworkManager.ServerClientId);
+            }
+            else
+            {
+                foreach (var clientId in NetworkManager.ConnectedClientsIds)
                 {
-                    foreach (var clientId in m_NetworkManager.ConnectedClientsIds)
+                    if (clientId != NetworkManager.ServerClientId)
                     {
-                        if (clientId != m_NetworkManager.ServerClientId)
-                        {
-                            command.TargetClientIds.Add(clientId);
-                        }
+                        clientList.Add(clientId);
                     }
                 }
+            }
+
+            return clientList;
+        }
+
+        internal void AddSpawn(SnapshotSpawnCommand command)
+        {
+            if (NumSpawns >= m_MaxSpawns)
+            {
+                Array.Resize(ref Spawns, 2 * m_MaxSpawns);
```

Comment:
Why are we not just using a list for this? Looks like we keep track of size and resize which is exactly what a list is for?

Reply:
That might be me coming from C++. Are C# list contiguous in memory ? If so, do they grow in an amortized linear fashion ? If both are yes, then sure, List would work then, too. 
What I wanted to avoid was triggering memory alloc at every `AddSpawn`. I'll learn on C# Lists first, then we can decide.

Reply:
Might not be too early to use NativeList<T>

Reply:
Lists in c# are contiguous in memory. They grow by doubling the size of their underlying buffer (and memory copy the old content). Same goes for NativeList.

Reply:
:+1: thanks, I really got confused by their naming.
MTT-1101 will address this.